app/models/marc_extractor.rb

```diff
@@ -0,0 +1,91 @@
+class MarcExtractor
+  attr_reader :marc, :tags, :subfields
+
+  def initialize(marc, tags)
+    @marc = marc
+    @tags = []
+    @subfields = {}
+    tags = tags.split(':') if tags.is_a? String
+
+    Array.wrap(tags).each do |tag|
+      t, subfields_spec = tag.to_s.scan(/\d{3}|\w+/)
```

Comment:
👀 this regex magic and mushing fields and subfields together (e.g. `%w(382abdnpsv)`) feels a little weird to me but... I guess we're only using it in 3 places.

Reply:
yeah; it's kinda funny, but matches up with what we're doing over in traject:

https://github.com/traject/traject/blob/master/lib/traject/marc_extractor.rb#L11-L13

If we ever want to add indicator support, maybe we'd just pull the traject parser in entirely 🤷‍♂️